app/controllers/devices_controller.rb

```diff
@@ -0,0 +1,48 @@
+class DevicesController < ApplicationController
+    def index
+       devices = Device.all
+       render json: devices, each_serializer:DeviceSerializer, status: :ok
+    end
+
+    def search
+        @device = Device.search(params[:search])
+        render json:@device
+    end
+
+    def create
+        @device = Device.create!(device_params)
+        render json:@device,each_serializer:DeviceSerializer, status: :ok
+    end
+
+    def destroy
+        device = find_params
+        device.destroy
+        render json: "device deleted", status: :ok  
+    end
+
+    def update
+        @device = find_params
+        @device.update(device_params)
+        render json:@device, status: :ok
+    end
+
+    def assigned_all
```

Comment:
rename 'assigned_all'  to 'assigned', so that the endpoint would be
/devices/assigned